python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/resources.py

```diff
@@ -97,129 +132,41 @@ class SlingTargetConnection(PermissiveConfig):
-                see https://docs.slingdata.io/sling-cli/running-tasks#target-options-tgt-options-flag-target.options-key
-                for details
-
-        Examples:
-            Sync from a source file to a sqlite database:
-
-            .. code-block:: python
-
-                sqllite_path = "/path/to/sqlite.db"
-                csv_path = "/path/to/file.csv"
-
-                @asset
-                def run_sync(context, sling: SlingResource):
-                    res = sling.sync(
-                        source_stream=csv_path,
-                        target_object="events",
-                        mode=SlingMode.FULL_REFRESH,
-                    )
-                    for stdout in res:
-                        context.log.debug(stdout)
-                    counts = sqlite3.connect(sqllitepath).execute("SELECT count(1) FROM events").fetchone()
-                    assert counts[0] == 3
-
-                source = SlingSourceConnection(
-                    type="file",
-                )
-                target = SlingTargetConnection(type="sqlite", instance=sqllitepath)
-
-                materialize(
-                    [run_sync],
-                    resources={
-                        "sling": SlingResource(
-                            source_connection=source,
-                            target_connection=target,
-                            mode=SlingMode.TRUNCATE,
-                        )
-                    },
-                )
+    """
 
-        """
-        yield from self._sync(
-            source_stream=source_stream,
-            target_object=target_object,
-            mode=mode,
-            primary_key=primary_key,
-            update_key=update_key,
-            source_options=source_options,
-            target_options=target_options,
-        )
-
-    @abstractmethod
-    def _sync(
-        self,
-        source_stream: str,
-        target_object: str,
-        mode: SlingMode = SlingMode.FULL_REFRESH,
-        primary_key: Optional[List[str]] = None,
-        update_key: Optional[str] = None,
-        source_options: Optional[Dict[str, Any]] = None,
-        target_options: Optional[Dict[str, Any]] = None,
-        encoding: str = "utf8",
-    ) -> Generator[str, None, None]:
-        """Runs a Sling sync from the given source table to the given destination table. Generates
-        output lines from the Sling CLI.
-        """
-        raise NotImplementedError()
+    source_connection: SlingSourceConnection
+    target_connection: SlingTargetConnection
 
+    @contextlib.contextmanager
+    def _setup_config(self) -> Generator[None, None, None]:
+        """Uses environment variables to set the Sling source and target connections."""
+        sling_source = _process_env_vars(dict(self.source_connection))
+        sling_target = _process_env_vars(dict(self.target_connection))
 
 @experimental
-class SlingResource(ConfigurableResource, _SlingSyncBase):
+class SlingResource(ConfigurableResource):
```

Comment:
is 'class SlingResource' defined twice in this file?

Reply:
Yea this is a bad merge conflict resolution. I'm working on resolving this.